### ISSUE TITLE
Snapshot API not working for ethernet devices

### ISSUE THREAD

**Issue by sumnerboy12** [austinscreations/OXRS-AC-TP32-ESP-FW#24]
`api/snapshot.bin` returns 2048 bytes of NULLs. The same release for WiFi devices works fine.

Tested on 0.2.1.

**Comment by moinmoin-sh** [comment]
I don't think this is a TP-FW related issue. The ethernet library seems to be less smart than the Wifi and can't handle payloads larger than 2k or needs some special code in your API lib to partition larger payloads(?). What you see in the buffer is probably the first ~700 (~2 rows) pixel of the black background which you selected [0,0,0] . If you would change this you would probably see the color you set the theme to.

**Comment by sumnerboy12** [comment]
I found this in `Ethernet.h`;

```
// By default, each socket uses 2K buffers inside the WIZnet chip.  If
// MAX_SOCK_NUM is set to fewer than the chip's maximum, uncommenting
// this will use larger buffers within the WIZnet chip.  Large buffers
// can really help with UDP protocols like Artnet.  In theory larger
// buffers should allow faster TCP over high-latency links, but this
// does not always seem to work in practice (maybe WIZnet bugs?)
//#define ETHERNET_LARGE_BUFFERS
```

I have been playing around enabling this and it hasn't made any difference. I even added `Serial.print()` lines to confirm the internal buffers in `w5100.cpp` are being increased (which they are).

I haven't time to look into this further at the moment so will leave this open.
